test/e2e_node/critical_pod_test.go

```diff
@@ -78,11 +78,18 @@ var _ = framework.KubeDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:C
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			})
+
 			nonCriticalBestEffort := getTestPod(false, bestEffortPodName, v1.ResourceRequirements{})
+			capacity := getNodeCPUAndMemoryCapacity(f)
+			used := v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("200m"),
+				v1.ResourceMemory: resource.MustParse("200Mi"),
+			}
+			available := getAvailableCPUAndMemoryCapacity(capacity, used)
 			criticalPod := getTestPod(true, criticalPodName, v1.ResourceRequirements{
 				// request the entire resource capacity of the node, so that
 				// admitting this pod requires the other pod to be preempted
-				Requests: getNodeCPUAndMemoryCapacity(f),
+				Requests: available,
```

Comment:
I think the usage of `getNodeCPUAndMemoryCapacity` is intentional. If a non-critical pod requests all the resources it will stay in pending, but a system critical pod should evict all other pods and get scheduled.

Reply:
when `getNodeCPUAndMemoryCapacity `, critical pod will not be admited because of OutOfCPU or OutOfMem

Reply:
Because e2e_node.test only start kubelet and apiserver without scheduler which must Preemption .

Reply:
That is false.  This is not a test for scheduler preemption, but rather a test for kubelet preemption, which will admit a critical pod even if resources do not exist on the node for it.

This should be changed back to getNodeCPUAndMemoryCapacity().  It would defeat the purpose of the test to subtract an amount from it.